packages/console/src/pages/SignInExperience/components/ManageLanguage/LanguageEditor/LanguageDetails.tsx

```diff
@@ -29,8 +30,9 @@ const LanguageDetails = () => {
 
   const { data: signInExperience } = useSWR<SignInExperience, RequestError>('/api/sign-in-exp');
 
-  const { languages, selectedLanguage, setIsDirty, setSelectedLanguage, stopAddingLanguage } =
-    useContext(LanguageEditorContext);
+  const { languages } = useUiLanguages();
+
+  const { selectedLanguage, setIsDirty, setSelectedLanguage } = useContext(LanguageEditorContext);
```

Comment:
Then do we still need the context?

Reply:
Yes, we need it; many states are still shared between components.